be/src/runtime/hdfs/hdfs_fs_cache.cpp

```diff
@@ -21,6 +21,23 @@
 
 namespace starrocks {
 
+// Try to get azure cloud properties from FSOptions, if azure credentials not existed, return nullptr.
+// TODO(SmithCruise): Should remove when using azure cpp sdk
+static const std::vector<TCloudProperty>* get_azure_cloud_properties(const FSOptions& options) {
+    const TCloudConfiguration* cloud_configuration = nullptr;
+    if (options.cloud_configuration != nullptr) {
+        // This branch is used by data lake
+        cloud_configuration = options.cloud_configuration;
+    } else if (options.hdfs_properties() != nullptr && options.hdfs_properties()->__isset.cloud_configuration) {
+        // This branch is used by broker load
+        cloud_configuration = &options.hdfs_properties()->cloud_configuration;
+    }
+    if (cloud_configuration != nullptr && cloud_configuration->cloud_type == TCloudType::AZURE) {
+        return &cloud_configuration->cloud_properties;
```

Comment:
return temp address?

Reply:
`cloud_configuration` comes from `const FSOptions& options`, it's a reference from the upper function, so that will not return the temp address.